src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/MLDsa/MLDsaTestHelpers.cs

```diff
@@ -0,0 +1,473 @@
+                    return importPem(pem, password);
+                });
+            }, passwordTypeToTest);
+        }
+
+        internal delegate MLDsa ImportFromEncryptedPemCallback(string source, string password);
+        internal static void AssertImportFromEncryptedPem(
+            Action<ImportFromEncryptedPemCallback> callback,
+            EncryptionPasswordType passwordTypeToTest = EncryptionPasswordType.All)
+        {
+            if ((passwordTypeToTest & EncryptionPasswordType.Char) != 0)
+            {
+                callback(static (string pem, string password) => MLDsa.ImportFromEncryptedPem(pem, password));
+                callback(static (string pem, string password) => MLDsa.ImportFromEncryptedPem(pem.AsSpan(), password));
+            }
+
+            if ((passwordTypeToTest & EncryptionPasswordType.Byte) != 0)
+            {
+                callback(static (string pem, string password) =>
+                    MLDsa.ImportFromEncryptedPem(pem, Encoding.UTF8.GetBytes(password)));
+                callback(static (string pem, string password) =>
+                    MLDsa.ImportFromEncryptedPem(pem.AsSpan(), Encoding.UTF8.GetBytes(password)));
+            }
+        }
+
+        internal static void AssertExportMLDsaPublicKey(Action<Func<MLDsa, byte[]>> callback)
+        {
+            callback(mldsa =>
+            {
+                byte[] buffer = new byte[mldsa.Algorithm.PublicKeySizeInBytes];
+                mldsa.ExportMLDsaPublicKey(buffer.AsSpan());
+                return buffer;
+            });
+
+            AssertExportSubjectPublicKeyInfo(exportSpki =>
+                callback(mldsa =>
+                    SubjectPublicKeyInfoAsn.Decode(exportSpki(mldsa), AsnEncodingRules.DER).SubjectPublicKey.Span.ToArray()));
+        }
+
+        internal static void AssertExportMLDsaSecretKey(Action<Func<MLDsa, byte[]>> callback) =>
+            AssertExportMLDsaSecretKey(callback, callback);
+
+        internal static void AssertExportMLDsaSecretKey(Action<Func<MLDsa, byte[]>> directCallback, Action<Func<MLDsa, byte[]>> indirectCallback)
+        {
+            directCallback(mldsa =>
+            {
+                byte[] buffer = new byte[mldsa.Algorithm.SecretKeySizeInBytes];
+                mldsa.ExportMLDsaSecretKey(buffer.AsSpan());
+                return buffer;
+            });
+
+            AssertExportPkcs8PrivateKey(exportPkcs8 =>
+                indirectCallback(mldsa =>
+                    Decode(exportPkcs8(mldsa)).ToArray()));
+
+            static ReadOnlyMemory<byte> Decode(ReadOnlyMemory<byte> source)
+            {
+                MLDsaPrivateKeyAsn decoded =
+                    MLDsaPrivateKeyAsn.Decode(
+                        PrivateKeyInfoAsn.Decode(source, AsnEncodingRules.DER).PrivateKey, AsnEncodingRules.DER);
+
+                if (decoded.ExpandedKey is null)
+                    throw TestException_NoExpandedKey;
```

Comment:
Why are you throwing a static exception instance?  That's going to make things weird if two tests throw it in parallel